In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt 
from pathlib import Path

DATA_DIR = Path('../data')
CLEAN_DATA_FILE = 'Cleaned_Dataset.csv'
FULL_DATA_PATH = DATA_DIR / CLEAN_DATA_FILE


# Basic Look

Head, tail, shape, data types and missingness. 

In [7]:
data = pd.read_csv(FULL_DATA_PATH)

In [14]:
data.shape

(1796, 13)

In [4]:
display(data.head())
display(data.tail())

Job Title  \
0        Systems and Data Analysts   
1  Business and Marketing Analysts   
2  Business and Technical Analysts   
3        Systems and Data Analysts   
4        Systems and Data Analysts   

                                            Job Info          Position  \
0  Binance Accelerator Program - Data Analyst (Risk)      Risk Analyst   
1                                   Business Analyst  Business Analyst   
2                          Geophysicist/Data Analyst      Data Analyst   
3                Business Intelligence Data Engineer     Data Engineer   
4               BUSINESS INTELLIGENCE SPECIALIST, FT   Data Specialist   

                        Employer     City Province  \
0                        Binance   Remote    Undef   
1  Canadian Nuclear Laboratories   Remote    Undef   
2      Sander Geophysics Limited   Ottawa       ON   
3          Maximus Services, LLC  Toronto       ON   
4          Niagara Health System  Niagara       ON   

                                               Skill Seniority  Work Type  \
0                                        Python, Sql       ANY     Remote   
1                          Power Bi, Power BI, Excel       ANY     Remote   
2                                              Undef       ANY  In-Person   
3  Fabric, Power BI, Sql, Machine Learning, Genes...       ANY  In-Person   
4                          Azure, Power BI, SQL, Aws    Senior  In-Person   

  Industry Type  Min_Salary  Max_Salary  Avg_Salary  
0        Others    76000.00     89440.0   82720.000  
1        Others    65000.00     87000.0   76000.000  
2    Technology    64123.59     86600.0   75361.795  
3       Service    87875.00    105000.0   96437.500  
4    Healthcare   115211.20    130332.8  122772.000

Job Title  \
1791        Business and Marketing Analysts   
1792  Senior Business Intelligence Analysts   
1793               Senior Business Analysts   
1794      Senior Supply Chain Data Analysts   
1795  Senior Business Intelligence Analysts   

                                       Job Info          Position  \
1791                        IM Business Analyst  Business Analyst   
1792  Risk Analyst, Third Party Risk Management      Risk Analyst   
1793              Business Intelligence Analyst        BI Analyst   
1794        Analyst, Supply Chain Replenishment           Analyst   
1795                   Project Controls Analyst           Analyst   

                      Employer           City Province  \
1791                     Messa         Ottawa       ON   
1792                     Aviva        Markham       ON   
1793                      Clio      Vancouver       BC   
1794  Loblaw Companies Limited       Brampton       ON   
1795             Suncor Energy  Fort McMurray       AB   

                                                  Skill Seniority  Work Type  \
1791                           Information architecture       ANY  In-Person   
1792  SQL, Operational risk management, Vendor manag...       ANY  In-Person   
1793  Power BI, Data modeling, Business intelligence...       ANY  In-Person   
1794      SQL, SAP, Computer skills, Marketing, Workday       ANY  In-Person   
1795       Investment, Power BI, SQL, SAP, Office Suite       ANY  In-Person   

     Industry Type  Min_Salary  Max_Salary  Avg_Salary  
1791        Others     67000.0    102000.0     84500.0  
1792        Others     94000.0    110000.0    102000.0  
1793        Others     86000.0     94000.0     90000.0  
1794    Technology     52000.0     59000.0     55500.0  
1795        Energy     72000.0    153000.0    112500.0

In [12]:
sum(data.iloc[:,-1] == ( data.iloc[:,-3] + data.iloc[:, -2] ) / 2)/len(data)

0.9894209354120267

In [13]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1796 entries, 0 to 1795
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Job Title      1796 non-null   object 
 1   Job Info       1796 non-null   object 
 2   Position       1796 non-null   object 
 3   Employer       1796 non-null   object 
 4   City           1796 non-null   object 
 5   Province       1796 non-null   object 
 6   Skill          1796 non-null   object 
 7   Seniority      1796 non-null   object 
 8   Work Type      1796 non-null   object 
 9   Industry Type  1796 non-null   object 
 10  Min_Salary     1796 non-null   float64
 11  Max_Salary     1796 non-null   float64
 12  Avg_Salary     1796 non-null   float64
dtypes: float64(3), object(10)
memory usage: 182.5+ KB


## Comments
- 1796 entries and 13 columns.
- Data seems to have no errors in the beginning and end
    - `Job Info` and `Skill` could be reformatted to remove punctuation and lower case for further language processing.
    - Could one hot encode `Skill` for the most popular 20 skills. 
        - PROS: Each skill can become a useable feature in modelling. 
        - CONS: May reduce the dataset substrantially. 
    - Could also just drop `Job Info` seems like the information is captured by `Job Title` and `Position`.
- Avg salary is just the mid point between the min and max estimates of salary.
    - Is this best way to go about this?
    
- Seems to have no null values. 


In [17]:
data.drop(labels=["Job Info"], axis=1, inplace=True)

In [18]:
data.head()

Job Title          Position  \
0        Systems and Data Analysts      Risk Analyst   
1  Business and Marketing Analysts  Business Analyst   
2  Business and Technical Analysts      Data Analyst   
3        Systems and Data Analysts     Data Engineer   
4        Systems and Data Analysts   Data Specialist   

                        Employer     City Province  \
0                        Binance   Remote    Undef   
1  Canadian Nuclear Laboratories   Remote    Undef   
2      Sander Geophysics Limited   Ottawa       ON   
3          Maximus Services, LLC  Toronto       ON   
4          Niagara Health System  Niagara       ON   

                                               Skill Seniority  Work Type  \
0                                        Python, Sql       ANY     Remote   
1                          Power Bi, Power BI, Excel       ANY     Remote   
2                                              Undef       ANY  In-Person   
3  Fabric, Power BI, Sql, Machine Learning, Genes...       ANY  In-Person   
4                          Azure, Power BI, SQL, Aws    Senior  In-Person   

  Industry Type  Min_Salary  Max_Salary  Avg_Salary  
0        Others    76000.00     89440.0   82720.000  
1        Others    65000.00     87000.0   76000.000  
2    Technology    64123.59     86600.0   75361.795  
3       Service    87875.00    105000.0   96437.500  
4    Healthcare   115211.20    130332.8  122772.000

# Exploring `Job Title`

In [22]:
np.round(data['Job Title'].value_counts()/len(data)*100, 2)

Job Title
Senior Supply Chain Data Analysts        14.59
Senior Business Intelligence Analysts    12.75
Senior Business Analysts                 11.92
Business Systems Analysts                10.41
Systems and Data Analysts                10.30
Senior Data Analysts                      9.47
Financial and Operational Analysts        9.08
Business and Technical Analysts           8.91
Business and Marketing Analysts           8.85
Database Analysts                         3.73
Name: count, dtype: float64

## Comments

- Almost 50 percent of the job titles in the dataset have Senior in them.
    - Could this signal the difficulty in getting entry level positions
    - Need to explore `Seniority` to get a better understanding. 


# Exploring `Position`

In [24]:
np.round(data['Position'].value_counts()/len(data)*100, 2)

Position
Analyst                 43.43
Business Analyst        18.21
Data Analyst            12.75
System Analyst          10.13
Data Engineer            2.28
BI Analyst               1.89
Risk Analyst             1.56
Developer                1.56
Lead                     1.34
Data Specialist          1.22
Data Consultant          0.89
Intern                   0.78
Logistic Analyst         0.61
Programmer               0.50
Manager                  0.45
Quantitative Analyst     0.39
Data Associate           0.33
Financial Analyst        0.33
Data Officer             0.28
Director                 0.17
Coordinator              0.17
Other                    0.17
Data Architect           0.17
Statistician             0.17
Data Scientist           0.11
Executive                0.06
Trainer                  0.06
Name: count, dtype: float64

## Comments

- Mostly related to analyst position. 
- What is a `Trainer` position?